### ISSUE TITLE
tf.nn.conv3d_transpose operation with data_format='NCDHW'

### ISSUE THREAD

**Issue by d-acharya** [tensorflow/tensorflow#18460]
### System information
Irrelevant.

### Describe the problem
In implementation of "tf.nn.conv3d_transpose" operation with "data_format='NCDHW'",
the shape compatibility of "output shape" and "kernel size" check is not carried out correctly. (However, the check is fine in tf.nn.conv2d_transpose)

### Source code / logs
`    if isinstance(output_shape, (list, np.ndarray)):
      # output_shape's shape should be == [5] if reached this point.
      if not filter.get_shape()[3].is_compatible_with(output_shape[4]):
        raise ValueError(
            "output_shape does not match filter's output channels, "
            "{} != {}".format(output_shape[4],
                              filter.get_shape()[3]))`

should be changed to 

`    if isinstance(output_shape, (list, np.ndarray)):
      # output_shape's shape should be == [5] if reached this point.
      if not filter.get_shape()[3].is_compatible_with(output_shape[axis]):
        raise ValueError(
            "output_shape does not match filter's output channels, "
            "{} != {}".format(output_shape[axis],
                              filter.get_shape()[3]))`

more specifically, output_shape[4] to output_shape[axis].

**Comment by yongtang** [comment]
Thanks @d-acharya. Created a PR #18465 and add a test case to address the issue.
